### PR TITLE
clean up the broken cors stuff

### PR DESCRIPTION
**Pull request by taylorking**

### CODE REVIEW

File: scripts/lua/cors.lua [hunk 1]
```diff
@@ -25,24 +25,33 @@ local _M = {}
 local request = require "lib/request"
 
 function _M.processCall(resourceConfig)
-  if resourceConfig.cors ~= nil then
-    _M.setCorsHeaders(resourceConfig.cors.origin, resourceConfig.cors.methods)
-    if ngx.req.get_method() == "OPTIONS" then
+  ngx.var.cors_origins = resourceConfig.cors.origin
+  ngx.var.cors_methods = resourceConfig.cors.methods
+  if resourceConfig.cors ~= nil then
```
Comment (mhamann) [inline scripts/lua/cors.lua]: Any reason not to make this a single `if` statement?

File: scripts/lua/cors.lua [hunk 2]
```diff
@@ -25,24 +25,33 @@ local _M = {}
...
-  if corsOrigin ~= nil then
-    if corsOrigin == 'false' then
+function _M.replaceHeaders()
+  if ngx.var.cors_origins ~= nil then
+    if ngx.var.cors_origins == 'true' then
+      ngx.header['Access-Control-Allow-Origin'] = '*'
+      if ngx.var.cors_methods ~= nil then
+        ngx.header['Access-Control-Allow-Methods'] = ngx.var.cors_methods
+      else
+        ngx.header['Access-Control-Allow-Methods'] = 'GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS'
```
Comment (mhamann) [inline scripts/lua/cors.lua]: Instead of `else`ing this, why not just make it the default and override it if `cors_methods ~= nil`?

Comment (taylorking) [inline scripts/lua/cors.lua]: done

File: scripts/lua/cors.lua [hunk 3]
```diff
@@ -25,25 +25,28 @@ local _M = {}
 local request = require "lib/request"
 
 function _M.processCall(resourceConfig)
-  if resourceConfig.cors ~= nil then
-    _M.setCorsHeaders(resourceConfig.cors.origin, resourceConfig.cors.methods)
-    if ngx.req.get_method() == "OPTIONS" then
-      request.success(200)
-    end
+  ngx.var.cors_origins = resourceConfig.cors.origin
+  ngx.var.cors_methods = resourceConfig.cors.methods
```
Comment (alexsong93) [inline scripts/lua/cors.lua]: I think these two should go inside the if statement below because it will fail if `resourceConfig.cors` is not defined

Comment (taylorking) [inline scripts/lua/cors.lua]: good point

**Pull request merged**